// Copyright 2021 Google LLC. All Rights Reserved.
// 
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
// 
//     http://www.apache.org/licenses/LICENSE-2.0
// 
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
syntax = "proto3";

package dcl;

import "proto/connector/sdk.proto";
import "proto/empty.proto";

enum ServiceusageServiceusageServiceStateEnum {
  ServiceusageServiceusageServiceStateEnumNO_VALUE_DO_NOT_USE = 0;
  ServiceusageServiceusageServiceStateEnumDISABLED = 1;
  ServiceusageServiceusageServiceStateEnumENABLED = 2;
}

message ServiceusageServiceusageService {
  string name = 1;
  ServiceusageServiceusageServiceStateEnum state = 2;
  string project = 3;
}

message ApplyServiceusageServiceusageServiceRequest {
  ServiceusageServiceusageService resource = 1;
  repeated LifecycleDirective lifecycle_directives = 2;
  string service_account_file = 3;
}

message DeleteServiceusageServiceusageServiceRequest {
  string service_account_file = 1;
  ServiceusageServiceusageService resource = 2;
}

message ListServiceusageServiceusageServiceRequest {
  string service_account_file = 1;
  string Project = 2;
}

message ListServiceusageServiceusageServiceResponse {
  repeated ServiceusageServiceusageService items = 1;
}
service ServiceusageServiceusageServiceService {
  rpc ApplyServiceusageServiceusageService(ApplyServiceusageServiceusageServiceRequest) returns (ServiceusageServiceusageService);
  rpc DeleteServiceusageServiceusageService(DeleteServiceusageServiceusageServiceRequest) returns (google.protobuf.Empty);
  rpc ListServiceusageServiceusageService(ListServiceusageServiceusageServiceRequest) returns (ListServiceusageServiceusageServiceResponse);
}
